### ISSUE TITLE
Improve source-map generation in Fable

### ISSUE THREAD

**Issue by MangelMaxime** [fable-compiler/Fable#2013]
Right now the source-map generation seems to cause a lot of problems when people try to debug the F# code.

We should look at this problem and see what can be improved.

**Comment by 7sharp9** [comment]
Im happy to investigate/help if you let me know some pointers or thing to read up on.

**Comment by alfonsogarciacaro** [comment]
Actually after #1866 I've noticed the debugging experience is much better, though it's true that things like pipes and computation expressions are confusing when stepping, because the generated code doesn't look like the original one.

@matthid also provided a tool to inspect the source maps in the mentioned PR. Can you use the tool for the specific examples you may have and see how they can be improved?

**Comment by alfonsogarciacaro** [comment]
Closing in favor of #2166 as currently Fable 3 doesn't generate any source-map at all :)
